src/protocol/fetch.rs

```diff
@@ -145,13 +144,15 @@ where
 
 impl<'a, 'b, 'c> super::ResponseParser for ResponseParser<'a, 'b, 'c> {
     type T = Response;
-    fn parse(&self, response: Vec<u8>) -> Result<Self::T> {
+    fn parse(&self, response: Vec<u8>) -> Result<Self::T, Error> {
         Response::from_vec(response, self.requests, self.validate_crc)
     }
 }
 
 // ~ helper macro to aid parsing arrays of values (as defined by the
 // Kafka protocol.)
+// zlb: why do this as a macro vs a function call?
```

Comment:
@estk I had this question generally. For most of the error handling stuff I changed them to functions, but I'm realizing now it may be faster to do macros. Perhaps I am supposed to explicitly call for those functions to be inlined? Not sure what the author's intent was with a lot of the macros. I had assumed that error chain forced it to be macros since it relied on macros to do to the actual error result handling.